sqlx-core/src/connection.rs

```diff
@@ -189,6 +189,9 @@ pub trait ConnectOptions: 'static + Send + Sync + FromStr<Err = Error> + Debug +
     /// Parse the `ConnectOptions` from a URL.
     fn from_url(url: &Url) -> Result<Self, Error>;
 
+    /// Get the URL set for the current `ConnectOptions`.
+    fn get_url(&self) -> Url;
```

Comment:
This is a breaking change because it adds a new required method to a public trait, which would affect third-party driver crates.

Instead, it could just be an inherent method that each `ConnectOptions` type can implement independently.

Alternatively, it could default to a stub implementation, though this is less idiomatic (and I was planning to focus on 0.8.0 next anyway):

```rust
    /// <include docs suggestion above>
    ///
    /// ### Panics
    /// This defaults to `unimplemented!()`.
    /// 
    /// Individual drivers should override this to implement the intended behavior.
    fn to_url(&self) -> Url {
        unimplemented!()
    }
```

Reply:
Naming nit: `get_*` implies returning a reference in Rust. IntelliJ-Rust actually has a lint warning for this.

`to_url` would be more idiomatic.

Reply:
On the other hand, `to_url` also sort of implies a lossless conversion.

Maybe `to_url_lossy` to make it clear.

Reply:
I don't have a strong use case for having the method in the trait, other than it being a "nice to have" kind of thing, and potentially enforcing implementation of the method when implementing the trait, for uniformity sake.

Either way I'd be happy to have a way to get URLs from the main databases, as that is immensely useful for passing it on to a framework like Rocket in testing, for changing the configuration of the service for each test.